frontend/ui/widget/numberpickerwidget.lua

```diff
@@ -146,7 +146,49 @@ function NumberPickerWidget:init()
                             text = _("OK"),
                             is_enter_default = true,
                             callback = function()
-                                local input_value = tonumber(input_dialog:getInputText())
+                                local input_text = input_dialog:getInputText()
+                                local input_value = tonumber(input_text)
+                                local turn_off_checks = false
+
+                                -- if the first character of the input string is ":" turn of min-max checks
+                                if input_text:match("^:") and G_reader_settings:isTrue("debug_verbose") then
+                                    turn_off_checks = true -- turn off checks
+                                    input_text = input_text:gsub("^:", "") -- shorten input
+                                    input_value = tonumber(input_text) -- try to get value
+                                end
+
+                                -- if the input text starts with `=` try to evaluate the expression
+                                -- only allow `math.*` and no other functions to be safe here.
+                                if input_text:match("^=") then
+                                    input_text = input_text:gsub("^=", "return ")
+                                    -- run code under environment
+                                    local function run_lua_protected_string(code)
+                                        -- make environment
+                                        local env = {math = math}
+                                        local func, dummy = loadstring(code)
```

Comment:
LuaJIT supports the 5.2 extension where you can directly pass an env to [`load`](https://www.lua.org/manual/5.2/manual.html#pdf-load)

Reply:
Yep, and I will have to make the sandbox more restictive -> http://lua-users.org/wiki/SandBoxes

I have a good (not ov..........red) idea for this. But no time right now (maybe next week )

Reply:
Well, we don't need to be super sandboxed, it's only the user or his baby child inputing stuff. And his baby child doing long-press and deleting books is more serious and more probable that she entering some valid lua one-liners :)

Reply:
Well thats true, but restricting chars to only show up after  "math." and prohibit everything except +-*/()^.% digits and chars will be a one- or two-liner.
(I assume that there is no need for bool operators in this context :) )

With that strategy we have a supersafe sandbox ;-)

Reply:
Another one cent: To have a safe sandbox might be an issue on Android - FDroid

Reply:
In any case, the process involved in dealing with this is definitely fun and interesting (err, to me, at least).

(And, yeah, I'd have hard-veto'ed this without some sort of sanitization on user input anyway ^^).

Reply:
(hard to resist not throwing the o-word)
(If this starts with 40 lines for an hidden uneeded "feature", and it ends up being 80...)

Reply:
(I swear  maximum 5 lines of code more.
But I did not say how long the lines will be :-)
@poire-z Would you tell me the O-word? Does it end with  th? ;-) )

Reply:
(Oh-my-goth !)